### PR TITLE
Support iOS devices

### PR DESCRIPTION
**Pull request by ericlewis**
Optimizes the UI for working across iPad / iPhone / macOS

### CODE REVIEW

**Review by pcuenca — COMMENTED** [review]
I haven't tried to optimize for iOS yet, but it's cool to start working on iOS integration!

This PR looks great in terms of UI. However, it won't be possible to read the model file on iOS unless we use `startAccessingSecurityScopedResource()` on the file selected by the user in the file browser. Would you like to address that, or would you prefer that I submit it as a separate PR first?

**Comment by ericlewis** [comment]
> I haven't tried to optimize for iOS yet, but it's cool to start working on iOS integration!
> 
> This PR looks great in terms of UI. However, it won't be possible to read the model file on iOS unless we use `startAccessingSecurityScopedResource()` on the file selected by the user in the file browser. Would you like to address that, or would you prefer that I submit it as a separate PR first?

Different PR would probably be a good idea. Works fine in simulator, never heard of security scoped resource on device 👁️

**Comment by ericlewis** [comment]
oh and I can address it if you like!

**Comment by pcuenca** [comment]
> oh and I can address it if you like!

Amazing, thank you! Let me know if you need help.

**Comment by ericlewis** [comment]
@pcuenca I added the security scope handling in this PR simply because it was easy enough. The LLaMA 2 model is too big for my iPhone (apparently) and the distilgpt2 mlpackage uses token_scores which isn't supported by swift-transformers yet.. *but* it appears that the prediction does successfully go through on device.

**Comment by pcuenca** [comment]
> distilgpt2 mlpackage uses token_scores which isn't supported by swift-transformers yet

Sorry about that, I'll upload a small compatible model later.

**Comment by ericlewis** [comment]
> > distilgpt2 mlpackage uses token_scores which isn't supported by swift-transformers yet
> 
> Sorry about that, I'll upload a small compatible model later.

modified swift-transformers to sort of support it, so no worries. rearranging UI after testing on device to feel more natural and will let you know when finished. changing to draft.

File: SwiftChat/ModelLoader.swift [hunk 1]
```diff
@@ -15,6 +15,11 @@ class ModelLoader {
         
     static func load(url: URL?) async throws -> LanguageModel {
         if let url = url {
+            
+            guard url.startAccessingSecurityScopedResource() else {
+                fatalError("Failed to access the model file due to security restrictions. Please ensure the app has the necessary permissions and the file is not in a restricted location.")
+            }
+
```
Comment (pcuenca) [inline SwiftChat/ModelLoader.swift:22]: Nice! I'd suggest we just add the call to the previous as an additional case in the previous `let` statement, remove the `fatalError` and use a `defer` block to call `url.stopAccessingSecurityScopedResource()` – I think it's a bit clearer if it's close to the `start...`.

**Comment by ericlewis** [comment]
https://github.com/huggingface/swift-chat/assets/674503/6a8c4159-5d5f-4054-bf6c-141ce5f958a2

iPhone now defaults to a more compact interface as demonstrated above. iPad and macOS share the same layouts.

**Comment by ericlewis** [comment]
@pcuenca okie dokie, this should be good now! I have tested on iPad, iPhone, and macOS devices.

File: SwiftChat/ModelLoader.swift [hunk 2]
```diff
@@ -14,7 +14,11 @@ class ModelLoader {
     static let lastCompiledModel = models / "last-model.mlmodelc"
         
     static func load(url: URL?) async throws -> LanguageModel {
-        if let url = url {
+        defer {
+            url?.stopAccessingSecurityScopedResource()
+        }
+        
+        if let url = url, url.startAccessingSecurityScopedResource() {
```
Comment (pcuenca) [inline SwiftChat/ModelLoader.swift:21]:
```suggestion
        if let url = url, url.startAccessingSecurityScopedResource() {
            defer {
                url?.stopAccessingSecurityScopedResource()
            }
```

Just a minor thing, but I'm not sure it's ok to call `stopAccessingSecurityScopedResource` if `start...` failed.

**Review by pcuenca — COMMENTED** [review]
Great work, I like it a lot!

This is almost ready to go, could you please take a look at the new comments and see if we can address them? 🙏

**Review by pcuenca — APPROVED** [review]
Thanks a lot, @ericlewis! Merging now, feel free to open a new PR for the increased memory capability :)

**Pull request merged**